app/Http/Controllers/Api/VolumeController.php

```diff
@@ -211,34 +211,43 @@ public function clone(CloneVolume $request)
     }
 
     /**
-     * Return volume's annotation timestamps of newest annotations
+     * Return file ids which are sorted by annotations.created_at
      *
      * @param int $id
      * @return object
-     * @api {get} volumes{/id}/files/annotation-timestamps Get the newest annotation timestamps of the volume
+     * @api {get} volumes{/id}/files/annotation-timestamps Get sorted file ids of volume
      * @apiGroup Volumes
      * @apiPermission projectMember
      *
      * @apiParam {Number} id The volume ID.
      *
      * @apiSuccessExample {json} Success response:
      * {
-     *  1: "2024-08-23T07:06:31.000000Z",
-     *  2: "2024-08-23T06:40:29.000000Z",
-     *  3: "2024-08-23T06:40:35.000000Z",
+     *  1: 1,
+     *  2: 2,
+     *  3: 3,
      * }
      *
      */
-    public function getAnnotationTimestamps($id)
+    public function getFileIdsSortedByAnnotationTimestamps($id)
     {
         $volume = Volume::findOrFail($id);
         $this->authorize('access', $volume);
 
-        $timestamps = $volume->files
-            ->flatMap(fn ($file) => $file->annotations)
-            ->groupBy('file_id')
-            ->map(fn ($annotations) => $annotations->max('created_at'));
+        if ($volume->isImageVolume()) {
+            $ids = $volume->files()
+                ->leftJoin('image_annotations', 'images.id', "=", "image_id")
+                ->groupBy('images.id')
+                ->selectRaw('images.id, max(created_at) as created_at');
+        } else {
+            $ids = $volume->files()
+                ->leftJoin('video_annotations', 'videos.id', "=", "video_id")
+                ->groupBy('videos.id')
+                ->selectRaw('videos.id, max(created_at) as created_at');
+        }
```

Comment:
I always specify the full table and column names. While it works here, it could fail in the future if e.g. a created_at` column was added to `images` or `annotations`.

```suggestion
            $ids = $volume->files()
                ->leftJoin('image_annotations', 'images.id', "=", "image_annotations.image_id")
                ->groupBy('images.id')
                ->selectRaw('images.id, max(image_annotations.created_at) as created_at');
        } else {
            $ids = $volume->files()
                ->leftJoin('video_annotations', 'videos.id', "=", "video_annotations.video_id")
                ->groupBy('videos.id')
                ->selectRaw('videos.id, max(video_annotations.created_at) as created_at');
        }
```